python/activator/middleware_interface.py

```diff
@@ -694,23 +715,19 @@ def _export_calib_associations(self, calib_collection, datasets):
             certified in ``calib_collection`` in the central repo, and must
             exist in the local repo.
         """
-        dataset_types = {ref.datasetType for ref in datasets}
-        associations = {}
-        for dataset_type in dataset_types:
-            associations.update(
-                (a.ref, a) for a in self.central_butler.registry.queryDatasetAssociations(
-                    dataset_type,
-                    calib_collection,
-                    collectionTypes={CollectionType.CALIBRATION},
-                    flattenChains=True
-                )
-            )
-        for dataset in datasets:
-            association = associations[dataset]
-            # certify is designed to work on groups of datasets; in practice,
-            # the total number of calibs (~1 of each type) is small enough that
-            # grouping by timespan isn't worth it.
-            self.butler.registry.certify(association.collection, [dataset], association.timespan)
+        # VALIDITY-HACK: (re)define a calibration collection containing the
+        # latest calibs as of today. Already-cached calibs are still available
+        # from previous collections.
```

Comment:
Are the "already-cached calibs" still of interest because we're concerned the new set may be incomplete (e.g. maybe there were new flats, but no new biases) and hence we need to work down the collection chain to find a match?  

If so, an alternative might be to pass out the full set of matching calibs from `_export_calibs` (i.e. without `known_datasets` subtracted) and certify/associate all of those either directly into a TAGGED/CALIBRATION `<instrument>/calib`, instead of prepending to a chain.

Reply:
That's one reason (how often do curated calibs change?).

Another is that (for instruments that are not LATISS) this needs to download calibs any time the pod sees a new detector, and the previously downloaded calibs may still be needed to process future observations with the "old" detector(s).

I would prefer not to try to play even more cleverness games with `_filter_datasets`, because the odds are overwhelming that this will introduce a logic bug with one of the other dataset types.